### PR TITLE
SDK: Try wp.data with calypso state tree "rewrite"

### PR DESCRIPTION
**Pull request by ockham**
WIP. Alternative to #26838.

Includes #26944, so review & merge that PR before rebasing this one.

* Use existing Calypso actions, selectors, _and reducers_.
* Also used via `withSelect`/`withDispatch`
* Use `@wordpress/data`'s `RegistryProvider`
* Selectors need to be modified to use local rather than global state (i.e. change `state.subtree.something` to `state.something`)
* All uses of modified selectors need to be changed from `connect` to `withSelect`

To Test:
1. Start calypso in local dev mode as usual.
2. Go to `http://calypso.localhost:3000/me/account` and change the "Admin Color Scheme"
3. Behavior should be the same, but it's using `withSelect` now.

### CODE REVIEW

File: client/boot/project/wordpress-com.js [hunk 1]
```diff
@@ -74,38 +59,32 @@ export function setupMiddlewares( currentUser, reduxStore ) {
...
-				const state = reduxStore.getState();
-				return {
-					blog_id: getSelectedSiteId( state ),
-					calypso_section: getSectionName( state ),
-				};
-			} );
-			errorLogger.saveDiagnosticReducer( () => ( { tests: getSavedVariations() } ) );
-			analytics.on( 'record-event', ( eventName, eventProperties ) =>
-				errorLogger.saveExtraData( { lastTracksEvent: eventProperties } )
+	if ( config.isEnabled( 'catch-js-errors' ) && ! document.getElementById( 'primary' ) ) {
```
Comment (coderkevin) [inline client/boot/project/wordpress-com.js]: This code seems extraneous to this PR's purpose. If it's a separate improvement, would it make sense to put it in a standalone PR?

Comment (ockham) [reply client/boot/project/wordpress-com.js]: It's #26944 for separate review, but I'm leaving it included on this branch since it wouldn't work otherwise. (tl;dr this extraneous -- and obsolete -- empty `Layout` would leave `wpRegistry` `undefined` upon first render of `RegistryProvider`, and that breaks things quite badly).

**Comment by simison** [comment]
I suppose this can be closed?

**Comment by sirreal** [comment]
I'll close this, feel free to reopen if you'd like.
